test/powershell/engine/Cdxml/Cdxml.Tests.ps1

```diff
@@ -0,0 +1,268 @@
+$script:CimClassName = "PSCore_CimTest1"
+$script:CimNamespace = "root/default"
+$script:moduleDir = join-path $PSScriptRoot assets CimTest
+$script:deleteMof = join-path $moduleDir DeleteCimTest.mof
+$script:createMof = join-path $moduleDir CreateCimTest.mof
+
+$CimCmdletArgs = @{
+    Namespace = ${script:CimNamespace}
+    ClassName = ${script:CimClassName}
+    ErrorAction = "SilentlyContinue"
+    }
+
+$script:ItSkipOrPending = @{}
+
+function Test-CimTestClass {
+    (Get-CimClass @CimCmdletArgs) -ne $null
+}
+
+function Test-CimTestInstance {
+    (Get-CimInstance @CimCmdletArgs) -ne $null
+}
+
+Describe "Cdxml cmdlets are supported" -Tag CI,RequireAdminOnWindows {
+    BeforeAll {
+        $skipNotWindows = ! $IsWindows
+        if ( $skipNotWindows ) {
+            $script:ItSkipOrPending = @{ Skip = $true }
+            return
+        }
+
+        # start from a clean slate, remove the instances and the 
+        # class if they exist
+        if ( Test-CimTestClass ) {
+            if ( Test-CimTestInstance ) {
+                Get-CimInstance @CimCmdletArgs | Remove-CimInstance
+            }
+            $result = MofComp.exe $deleteMof
+            if ( $LASTEXITCODE -ne 0 ) {
+                $script:ItSkipOrPending = @{ Pending = $true }

```

Comment:
I thought about this, and while it might be a bug in the mof file or a problem with mofcomp, marking a test as pending doesn't get us off the hook - all pending tests must be resolved before we ship (and I do track our pending tests). However, I didn't want to invalidate the build because I couldn't even get to the actual test. I thought it was better to mark the test as pending rather than fail.